stdlib/public/runtime/Reflection.mm

```diff
@@ -299,6 +299,49 @@ AnyReturn swift_MagicMirrorData_objcValue(HeapObject *owner,
 #pragma clang diagnostic pop
 
 extern "C"
+void swift_OpaqueSummary(const Metadata *T, String *result) {
+  switch (T->getKind()) {
+    case MetadataKind::Class:
+    case MetadataKind::Struct:
+    case MetadataKind::Enum:
+    case MetadataKind::Optional:
+    case MetadataKind::Metatype:
+      new (result) String("");
+      break;
+    case MetadataKind::Opaque:
+      new (result) String("(Opaque Value)");
+      break;
+    case MetadataKind::Tuple:
+      new (result) String("(Tuple)");
+      break;
+    case MetadataKind::Function:
+      new (result) String("(Function)");
+      break;
+    case MetadataKind::Existential:
+      new (result) String("(Existential)");
+      break;
+    case MetadataKind::ObjCClassWrapper:
+      new (result) String("(Objective-C Class Wrapper)");
+      break;
+    case MetadataKind::ExistentialMetatype:
+      new (result) String("(ExistentialMetatype)");
+      break;
+    case MetadataKind::ForeignClass:
+      new (result) String("(Foreign Class)");
+      break;
+    case MetadataKind::HeapLocalVariable:
+      new (result) String("(Heap Local Variable)");
+      break;
+    case MetadataKind::HeapGenericLocalVariable:
+      new (result) String("(Heap Generic Local Variable)");
+      break;
+    case MetadataKind::ErrorObject:
+      new (result) String("(ErrorType Object)");
+      break;
+  }
+}
```

Comment:
Seems like swift_OpaqueSummary could be simplified by just returning a
const char* and letting the conversion happen in Swift code